dashboard/views/_positions/_ec/_secretary/_attendance.py

```diff
@@ -2,35 +2,40 @@
 
 import datetime
 
-from dashboard.models import Brother, ChapterEvent, Excuse
+from dashboard.models import Brother, RecruitmentEvent, Event, Excuse
 from dashboard.utils import verify_position, get_semester
 
-@verify_position(['Secretary', 'Vice President', 'President', 'Adviser'])
+
+@verify_position(['secretary', 'vice-president', 'president', 'adviser'])
 def secretary_attendance(request):
     """ Renders the secretary view for chapter attendance """
-    brothers = Brother.objects.exclude(brother_status='2').order_by('last_name')
-    events = ChapterEvent.objects.filter(semester=get_semester(), mandatory=True)\
-        .exclude(date__gt=datetime.date.today())
-    excuses = Excuse.objects.filter(event__semester=get_semester(), status='1')
-    events_excused_list = []
-    events_unexcused_list = []
+    brothers = Brother.objects.exclude(brother_status='2').order_by('last_name', 'first_name')
+    events = Event.objects.filter(semester=get_semester(), mandatory=True, date__lt=datetime.date.today())\
+        .exclude(pk__in=RecruitmentEvent.objects.all().values_list('pk', flat=True))
+    accepted_excuses = Excuse.objects.filter(event__semester=get_semester(), status='1', event__in=events)
+    brother_attendance = []
 
+    # For each brother, appends a tuple to brother_attendance that contains the number of events excused, unexcused,
+    # attended/excuse and events eligible for them to attend. The latter 2 are used to display the
+    # attendance fraction
     for brother in brothers:
+        events_eligible_list = events.filter(eligible_attendees=brother)
+        events_eligible = events_eligible_list.count()
+        events_attended = 0
         events_excused = 0
         events_unexcused = 0
-        for event in events:
-            if not event.attendees_brothers.filter(id=brother.id).exists():
-                if excuses.filter(brother=brother, event=event).exists():
-                    events_excused += 1
-                else:
-                    events_unexcused += 1
-        events_excused_list.append(events_excused)
-        events_unexcused_list.append(events_unexcused)
-
-    brother_attendance = zip(brothers, events_excused_list, events_unexcused_list)
+        for event in events_eligible_list:
+            if event.attendees_brothers.filter(id=brother.id).exists():
+                events_attended += 1
+            elif accepted_excuses.filter(brother=brother, event=event).exists():
+                events_excused += 1
+            else:
+                events_unexcused += 1
+        brother_attendance.append((brother, events_excused, events_unexcused, events_attended+events_excused, events_eligible))
```

Comment:
this is really similar to the recruitment chair attendance calculation...

Reply:
The two views are actually almost entirely the same with the exception of this line:
    events = Event.objects.filter(semester=get_semester(), mandatory=True, date__lt=datetime.date.today())\
        .exclude(pk__in=RecruitmentEvent.objects.all())

as opposed to
    events = RecruitmentEvent.objects.filter(semester=get_semester(), mandatory=True, date__lt=datetime.date.today())

but I'm not sure if there's any way to combine the two views besides using 'if' statements. Would that be worth it to do or should I just create a util to cover the overlap?

Reply:
You can pass in `Event` or `RecruitmentEvent` as a parameter

Reply:
to the view? I can only pass in strings (specifically strings that the url can hold) as parameter. There would have to be if statements in there to determine which would work. Additionally, in the prior it excludes recruitmentevents whereas in the latter it's using recruitment events as the starting point for the filter, no exclusion necessary

Reply:
eh whatever, as long as it works right now, thats good enough,